modules/standards/src/standard-user-definition.ts

```diff
@@ -1,57 +1,60 @@
-import { removePasswordFromResponseData } from './remove-password-from-response-data.js'
+import { StandardEntityDefinition } from './standard-entity-definition'
+import { encryptPasswordInRequestData } from './encrypt-password-in-request-data'
+import { removePasswordFromResponseData } from './remove-password-from-response-data'
 
-import type {
-  AuthSetting,
-  EntityMap,
+import {
+  GeneralReqResEntityMap,
+  Key,
   EntityClient,
-  EntityDefinition,
   UserDefinition,
-  LoginCommandOf,
   Session,
   AuthenticationResult,
-  RequestData,
-  ResponseData,
-  RestApiExecution,
-} from 'phenyl-interfaces'
+  GeneralRequestData,
+  GeneralResponseData,
+} from '@phenyl/interfaces'
 
-import type {
+import {
   EncryptFunction,
-} from '../decls/index.js.flow'
+  RestApiExecution,
+  AuthSetting,
+  LoginCommandOf
+} from '../decls/index'
 
-export type StandardUserDefinitionParams<M: EntityMap, A: AuthSetting> = {
+export type StandardUserDefinitionParams<M extends GeneralReqResEntityMap, A extends AuthSetting> = {
   entityClient: EntityClient<M>,
-  encrypt?: EncryptFunction,
-  accountPropName?: $Keys<$Values<M>> & $Keys<$ElementType<A, 'credentials'>>,
-  passwordPropName?: $Keys<$Values<M>> & $Keys<$ElementType<A, 'credentials'>>,
+  encrypt?: EncryptFunction<A>,
+  accountPropName?: Key<M[Key<M>]> & Key<A['credentials']>,
+  passwordPropName?: Key<M[Key<M>]> & Key<A['credentials']>,
   ttl?: number,
 }
 
-export class StandardUserDefinition<M: EntityMap = EntityMap, A: AuthSetting = AuthSetting> extends StandardEntityDefinition implements EntityDefinition, UserDefinition {
+// Q: is it necessary and possible to implement EntityDefinition and UserDefinition both?
+export class StandardUserDefinition<M extends GeneralReqResEntityMap, A extends AuthSetting> extends StandardEntityDefinition implements UserDefinition {
   entityClient: EntityClient<M>
-  encrypt: EncryptFunction
-  accountPropName: $Keys<$Values<M>> & $Keys<$ElementType<A, 'credentials'>>
-  passwordPropName: $Keys<$Values<M>> & $Keys<$ElementType<A, 'credentials'>>
+  encrypt: EncryptFunction<A>
+  accountPropName: Key<M[Key<M>]> & Key<A['credentials']>
+  passwordPropName: Key<M[Key<M>]> & Key<A['credentials']>
   ttl: number
 
   constructor(params: StandardUserDefinitionParams<M, A>) {
     super(params)
     this.entityClient = params.entityClient
     this.encrypt = params.encrypt || powerCrypt // TODO: pass salt string to powerCrypt
+    // @ts-ignore default
     this.accountPropName = params.accountPropName || 'account'
+    // @ts-ignore default
     this.passwordPropName = params.passwordPropName || 'password'
     this.ttl = params.ttl || 60 * 60 * 24 * 365 // one year
   }
 
-  async authentication<N: $Keys<M>>(loginCommand: LoginCommandOf<A, N>, session: ?Session): Promise<AuthenticationResult> { // eslint-disable-line no-unused-vars
+  // @ts-ignore @TODO: waiting for fix of interfaces
```

Comment:
could you explain what fix are required in `interfaces`?

Reply:
This class `StandardUserDefinition ` should implements `EntityDefinition` and `UserDefinition` interface at the same time.
But with the new `EntityDefinition` and `EntityDefinition` in the `phenyl/interfaces`, we cannot implement those two interfaces at the same time on the class `StandardUserDefinition `